Assets/GRPC.NET/Scripts/PushPullStream.cs

```diff
@@ -35,35 +28,33 @@ public override int Read(byte[] buffer, int offset, int count)
             var readLength = 0;
             lock (m_Buffer)
             {
-                while (!ReadAvailable(count))
-                    Monitor.Wait(m_Buffer);
+                while (!ReadAvailable()) Monitor.Wait(m_Buffer);
 
-                for (; readLength < count && Length > 0 && m_Buffer.Count > 0; readLength++)
-                {
+                for (; readLength < count && m_Buffer.Count > 0; readLength++)
                     buffer[readLength] = m_Buffer.Dequeue();
-                }
 
                 Monitor.Pulse(m_Buffer);
             }
 
             // BestHTTP expects us to return -1 when we have no data (but have not reached EOF yet).
-            if (readLength == 0 && !m_Closed)
-                return -1;
+            if (readLength == 0 && !m_Closed) return -1;
 
             return readLength;
         }
 
-        private bool ReadAvailable(int count)
+        private bool ReadAvailable()
         {
             if (m_Exception != null)
                 throw m_Exception;
 
             // Either we have data to read, or we got flushed (e.g. stream got closed)
             // or we are in non blocking read mode.
-            return Length >= count && m_Flushed || m_Closed || NonBlockingRead;
+            return m_Buffer.Count > 0 || m_Flushed || m_Closed || NonBlockingRead;
         }
 
-        public override long Length => m_Buffer.Count;
+        // Avoid sending content-length=0 (EOF) by using -2 for chunked encoding.
+        public override long Length => -2;
```

Comment:
We can leave this at -2 and Best HTTP will just stay in streaming/unbounded data mode, and will treat a -1 from `Read` as the EOF/EOS